scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -141,6 +142,95 @@ object DynamicTLSPeerGroup {
     }
   }
 
+  sealed abstract case class FramingConfig private (
+      maxFrameLength: Int,
+      lengthFieldOffset: Int,
+      lengthFieldLength: Int,
+      encodingLengthAdjustment: Int,
+      decodingLengthAdjustment: Int,
+      initialBytesToStrip: Int,
+      failFast: Boolean,
+      byteOrder: ByteOrder,
+      lengthIncludesLengthFieldLength: Boolean
+  )
+
+  object FramingConfig {
+    private val possibleLengthFieldLength = Set(1, 2, 3, 4, 8)
+
+    /**
+      *
+      * Configures framing format for all the peers in PeerGroup
+      *
+      *
+      * @param maxFrameLength the maximum length of the frame. If the length of the frame is greater than this value
+      *                       channel with generate DecodingError event.
+      * @param lengthFieldOffset the offset of the length field.
```

Comment:
Sorry, but it seems I cannot get the math of this offset. For example let's suppose a frame
`00 00 00 00 05` - can be decoded by framing config if 
1) `lengthFieldLength == 1`
2) `lengthFieldOffset == 4`
3) `lengthIncludesLengthFieldLength == true`
4) `maxFrameLength == 5`

And the data value in frame `00 00 00 00`. Is that correct?

Reply:
Yep standard case is `length + message` , and then in decoder we are stripping length bytes.